### PR TITLE
Fixed #27131 -- Passed proper string type to SMTP connection login

### PR DESCRIPTION
**Pull request by claudep**
Thanks slavugan@gmail.com for the report.

### CODE REVIEW

File: django/core/mail/backends/smtp.py [hunk 1]
```diff
@@ -64,7 +68,7 @@ def open(self):
                 self.connection.starttls(keyfile=self.ssl_keyfile, certfile=self.ssl_certfile)
                 self.connection.ehlo()
             if self.username and self.password:
-                self.connection.login(self.username, self.password)
+                self.connection.login(force_str(self.username), force_str(self.password))
```
Comment (timgraham) [inline django/core/mail/backends/smtp.py]: Should we add a comment that no coercion is needed after dropping Python 2?

File: tests/mail/tests.py [hunk 2]
```diff
@@ -1140,6 +1150,17 @@ def __init__(self, *args, **kwargs):
         self.active_lock = threading.Lock()
         self.sink_lock = threading.Lock()
 
+    def handle_accept(self):
+        if PY3:
```
Comment (timgraham) [inline tests/mail/tests.py]: Maybe it would be simpler to define this method only for Python 2?

File: tests/mail/tests.py [hunk 3]
```diff
@@ -1270,6 +1305,19 @@ def test_server_open(self):
         backend.close()
         self.assertTrue(opened)
 
+    def test_server_login(self):
+        """
+        Even if the Python SMTP server doesn't support authentication, this
```
Comment (timgraham) [inline tests/mail/tests.py]: ", the login process starts and the appropriate exception..." (chop "this checks that")

**Comment by timgraham** [comment]
It could be nice to note in the commit message that this fix is only required on Python 2 (as far as I understand).

**Comment by timgraham** [comment]
By the way, could a Django project work around this by declaring their credentials as bytes?

File: django/core/mail/backends/smtp.py (hunk 1 above)
Comment (claudep) [inline django/core/mail/backends/smtp.py]: I don't think it's useful. All `force_str` usage will have to be audited at the time, and most of them will indeed disappear.

**Comment by claudep** [comment]
I think you are right about the workaround. I still think it's worth committing, as the patch is testing a little bit deeper inside the SMTP connection process.

**Pull request closed without merging**